Library/ViewModels/ProjectPamphletViewModelTests.swift

```diff
@@ -516,6 +519,49 @@ final class ProjectPamphletViewModelTests: TestCase {
     }
   }
 
+  func testConfigurePledgeCTAView_reloadsUponRetryButtonTappedEvent() {
+    let config = Config.template |> \.features .~ [Feature.nativeCheckout.rawValue: true]
+    let project = Project.template
+    let projectFull = Project.template
+      |> \.id .~ 2
+      |> Project.lens.personalization.isBacking .~ true
+    let projectFull2 = Project.template
+      |> \.id .~ 3
+
+    let mockService = MockService(fetchProjectResponse: projectFull)
+
+    withEnvironment(apiService: mockService, config: config) {
+      self.configurePledgeCTAViewProject.assertDidNotEmitValue()
+      self.configurePledgeCTAViewIsLoading.assertDidNotEmitValue()
+
+      self.vm.inputs.configureWith(projectOrParam: .left(project), refTag: .discovery)
+      self.vm.inputs.viewDidLoad()
+
+      XCTAssertTrue(self.configurePledgeCTAViewProject.lastValue?.left == project)
+      self.configurePledgeCTAViewIsLoading.assertValues([true])
+
+      self.scheduler.advance()
+
+      XCTAssertTrue(self.configurePledgeCTAViewProject.lastValue?.left == projectFull)
+      self.configurePledgeCTAViewIsLoading.assertValues([true, true, false])
+    }
+
+    withEnvironment(
+      apiService: MockService(fetchProjectResponse: projectFull2),
+      config: config
+    ) {
+      self.vm.inputs.pledgeRetryButtonTapped()
+
+      XCTAssertTrue(self.configurePledgeCTAViewProject.lastValue?.left == projectFull)
+      self.configurePledgeCTAViewIsLoading.assertValues([true, true, false, true])
+
+      self.scheduler.advance()
+
+      XCTAssertTrue(self.configurePledgeCTAViewProject.lastValue?.left == projectFull2)
+      self.configurePledgeCTAViewIsLoading.assertValues([true, true, false, true, true, false])
```

Comment:
What do the multiple `true`s mean? Should we just add a `skipRepeats()`?

Reply:
This happens because `configurePledgeCTAView` combines the latest values of `freshProjectAndRefTagEvent` and `isLoading`. So whenever one signal updates, `configurePledgeCTAView` emits again. In this case, the second `true` after a `false` is simply the latest value of `isLoading` when `freshProjectAndRefTagEvent` emits a fresh project.